src/Arguments.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace Spatie\PhpUnitWatcher;
+
+use Spatie\PhpUnitWatcher\PhpUnit\Command as PhpUnitCommand;
+
+class Arguments
+{
+    use ArgumentAccessors;
+
+    const APPLICATION_OPTIONS = [];
+
+    protected $testFile;
+    protected $phpUnitOptions = [];
+    protected $applicationOptions = [];
+
+    /**
+     * Parses the given arguments string and instantiates a new object containing the arguments.
+     *
+     * @param  string $argumentsInput
+     * @return $this
+     */
+    public static function fromString($argumentsInput)
+    {
+        return (new static)->parse($argumentsInput);
+    }
+
+    /**
+     * Adds an argument if it does not yet exist, otherwise the argument value is replaced.
+     *
+     * In case $value is left empty, the argument will be treated as a boolean option. If the
+     * name and value are not separated by a space, pass the correct separator as a third
+     * parameter.
+     *
+     * Unless an option is registered (in this class) as an application option, we will assume
+     * the option is meant to be passed through to PHPUnit.
+     *
+     * @param  string $name
+     * @param  mixed $value
+     * @param  string $separator
+     */
+    public function addArgument($name, $value = null, $separator = '=')
+    {
+        $valueWithSeparator = is_null($value) ? $value : ['value' => $value, 'separator' => $separator];
+
+        if ($this->isApplicationOption($this->optionName($name))) {
+            $this->applicationOptions[$name] = $valueWithSeparator;
+
+            return;
+        }
+
+        $this->phpUnitOptions[$name] = $valueWithSeparator;
+    }
+
+    /**
+     * Returns a string containing all PHPUnit options with corresponding values in the correct format.
+     *
+     * @return string
+     */
+    public function phpUnitArguments()
+    {
+        $arguments = [];
+
+        foreach ($this->phpUnitOptions as $name => $optionValue) {
```

Comment:
Generally speaking I don't like `foreach` loops. Can you replace this with `array_map`?